assets/src/hooks/useDetour.ts

```diff
@@ -25,37 +28,76 @@ export enum DetourState {
   Finished,
 }
 
-export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
-  const [state, setState] = useState<DetourState>(DetourState.Edit)
+const selectFinishedState = (
+  snapshot: SnapshotFrom<typeof createDetourMachine>
+) => {
+  /*
+   * There's probably a better way to do this? Tags or maybe context?
+   * Tags seem more appropriate, but weird to manage Out-Of-Bounds state via tags.
+   * Maybe this entire API call could be moved to and managed by an actor
+   * combined with parallel child states within the state machine?
+   * -- https://stately.ai/docs/promise-actors */
```

Comment:
I'm a little conflicted on this but what I'm thinking is if we know that we can make a API call as soon as we reach the "Finished Drawing" state, then we could use XState promise actors to do the API call we want. 

There's two downsides
- I can't find many/any docs about using promise actors within another machine, especially when you don't want to block the application on that API call finishing, this is similar to my [other comment about nested machines](https://github.com/mbta/skate/pull/2607#discussion_r1611797335)
- If we restore to a state that has already "past" the "Finished Drawing" state, then it would be a little more difficult to find the right place to trigger the promise actor to do the API request.

Reply:
> Yeahhhhh it'd be nice to have a way to catch "any state that is past state X" in the flowchart instead of needing to check states individually. 🤔 Pretty simple here, but could get more complicated in the future?

Ideally https://github.com/mbta/skate/pull/2606 makes this better

Reply:
I made a few comments below, but I wonder if we could get rid of this check entirely (and thus get rid of the pain point) by storing the data a little differently

https://github.com/mbta/skate/pull/2607/files#r1612405667
https://github.com/mbta/skate/pull/2607/files#r1612399064

I might be way off-base here, but I have this hunch that the pain that we're experiencing here is really a symptom of trying to use the state machine _slightly_ skew to a "state machine-y" style - that is, we're trying to do too much "tell me about who you are", and not enough "tell me about what you can do".